src/Uuid/Uuid.php

```diff
@@ -0,0 +1,318 @@
+		string|Identifiable $seed,
+		Collection|null $context = null
+	): static {
+		if (is_string($seed) === true) {
+			return match (Str::before($seed, '://')) {
+				'page'   => new PageUuid(uuid: $seed, context: $context),
+				'file'   => new FileUuid(uuid: $seed, context: $context),
+				'site'   => new SiteUuid(uuid: $seed, context: $context),
+				'user'   => new UserUuid(uuid: $seed, context: $context),
+				// TODO: activate for uuid-block-structure-support
+				// 'block'  => new BlockUuid(uuid: $seed, context: $context),
+				// 'struct' => new StructureUuid(uuid: $seed, context: $context),
+				default  => throw new InvalidArgumentException('Invalid uuid uri:' . $seed)
+			};
+		}
+
+		return match (true) {
+			$seed instanceof Page
+				=> new PageUuid(model: $seed, context: $context),
+			$seed instanceof File
+				=> new FileUuid(model: $seed, context: $context),
+			$seed instanceof Site
+				=> new SiteUuid(model: $seed, context: $context),
+			$seed instanceof User
+				=> new UserUuid(model: $seed, context: $context),
+			// TODO: activate for uuid-block-structure-support
+			// $seed instanceof Block
+			// 	=> new BlockUuid(model: $seed, context: $context),
+			// $seed instanceof StructureObject
+			// 	=> new StructureUuid(model: $seed, context: $context),
+			default
+			=> throw new InvalidArgumentException('Uuid not supported for:' . get_class($seed))
+		};
+	}
+
+	/**
+	 * Generate a new ID string
+	 */
+	final public static function generate(int $length = 15): string
+	{
+		if (static::$generator !== null) {
+			return (static::$generator)($length);
+		}
+
+		return (new Nanoid())->generateId($length);
+	}
+
+	/**
+	 * Returns the UUID's id string
+	 */
+	public function id(): string
+	{
+		return $this->uri->host();
+	}
+
+	/**
+	 * Generator function that creates an index of
+	 * all identifiable model object globally
+	 */
+	public static function index(): Generator
+	{
+		yield from [];
+	}
+
+	/**
+	 * Merges local and global index generators
+	 * into one iterator
+	 * @internal
+	 */
+	final public function indexes(): Generator
+	{
+		yield from $this->context();
+		yield from static::index();
+	}
+
+	/**
+	 * Checks if a string resembles an UUID uri
+	 */
+	final public static function is(
+		string $string,
+		string|null $type = null
+	): bool {
+		if ($type !== null) {
+			return Str::startsWith($string, $type . '://');
+		}
+
+		// try to match any of the supported schemes
+		$pattern = sprintf('/^(%s):\/\//', implode('|', Uri::$schemes));
+		return preg_match($pattern, $string) === 1;
+	}
+
+	/**
+	 * Checks if Uuid has already been cached
+	 */
+	public function isCached(): bool
+	{
+		return Uuids::cache()->exists($this->key());
+	}
+
+	/**
+	 * Returns key for cache entry
+	 */
+	public function key(): string
+	{
+		$id = $this->id();
+
+		// for better performance when using a file-based cache,
+		// turn first two characters of the id into a directory
+		$id = substr($id, 0, 2) . '/' . substr($id, 2);
```

Comment:
Should use `Str` methods to be multi-byte safe in case a user uses custom IDs with Unicode characters.